testng-core/src/main/java/org/testng/xml/internal/TestNamesMatcher.java

```diff
@@ -43,13 +50,42 @@ public List<XmlSuite> getSuitesMatchingTestNames() {
     return cloneSuites;
   }
 
-  public void validateMissMatchedTestNames() {
-    List<String> tmpTestNames = Lists.newArrayList();
-    tmpTestNames.addAll(testNames);
-    tmpTestNames.removeIf(matchedTestNames::contains);
-    if (!tmpTestNames.isEmpty()) {
-      throw new TestNGException("The test(s) <" + tmpTestNames + "> cannot be found in suite.");
+  /**
+   * Do validation for testNames and notify users if any testNames are missed in suite. This method
+   * is also used to decide how to run test suite when test names are given. In legacy logic, if
+   * test names are given and exist in suite, then run them; if any of them do not exist in suite,
+   * then throw exception and exit. After ignoreMissedTestNames is introduced, if
+   * ignoreMissedTestNames is enabled, then any of the given test names exist in suite will be run,
+   * and print warning message to tell those test names do not exist in suite.
+   *
+   * @param ignoreMissedTestNames if true print warning message otherwise throw TestNGException for
+   *     missed testNames.
+   * @return boolean if ignoreMissedTestNames disabled, then return true if no missed test names in
+   *     suite, otherwise throw TestNGException; if ignoreMissedTestNames enabled, then return true
+   *     if any test names exist in suite, otehrwise (all given test names are missed) false.
+   */
+  public boolean validateMissMatchedTestNames(final boolean ignoreMissedTestNames) {
```

Comment:
I am not sure if we need to be setting up `final` for parameters or for local variables.

Reply:
any concern about using final here?

Reply:
The parameter is removed in new push. Instead, use the field and intialize in constructor. Therefore, the value should come from TestNG by option or VM argument.